pkg/operator/configobserver/oauth/observe_tokenconfig.go

```diff
@@ -0,0 +1,150 @@
+func ObserveAccessTokenInactivityTimeout(genericlisters configobserver.Listers, recorder events.Recorder, existingConfig map[string]interface{}) (ret map[string]interface{}, errs []error) {
+	errs = []error{}
+	observedConfig := map[string]interface{}{}
+	tokenConfigPath := []string{"oauthConfig", "tokenConfig"}
+	tokenInactivityTimeoutPath := append(tokenConfigPath, fieldAccessTokenInactivityTimeoutSeconds)
+	defer func() {
+		// Prune the observed config so that it only contains access token inactivity timeout field.
+		ret = configobserver.Pruned(ret, tokenInactivityTimeoutPath)
+	}()
+
+	listers, ok := genericlisters.(OAuthLister)
+	if !ok {
+		return existingConfig, append(errs, fmt.Errorf("failed to assert: given lister does not implement OAuth lister"))
+	}
+
+	existingAccessTokenInactivityTimeout, _, err := unstructured.NestedFieldCopy(existingConfig, tokenInactivityTimeoutPath...)
+	if err != nil {
+		// If failed to read existing config for access token inactivity timeout, continue execution with the
+		// hope to find the value for it in OAuth cluster config.
+		errs = append(errs, err)
+	}
+
+	var existingAccessTokenInactivityTimeoutFloat float64
+	if existingAccessTokenInactivityTimeout != nil {
+		if f, ok := existingAccessTokenInactivityTimeout.(float64); ok {
+			existingAccessTokenInactivityTimeoutFloat = f
+		}
+	}
+
+	oauthConfig, err := listers.OAuthLister().Get("cluster")
+	if err != nil {
+		// Failed to read OAuth cluster config.
+		if errors.IsNotFound(err) {
+			klog.Warning("oauth.config.openshift.io/cluster: not found")
+		}
+
+		if existingAccessTokenInactivityTimeoutFloat == 0 {
+			// As the nil value indicates that an attempt to read from existing config failed previously, return empty config.
+			return observedConfig, append(errs, err)
+		}
+		// Return whatever is present in existing config
+		return existingConfig, append(errs, err)
+	}
+
+	observedAccessTokenInactivityTimeout := oauthConfig.Spec.TokenConfig.AccessTokenInactivityTimeout.Seconds()
+	if observedAccessTokenInactivityTimeout != 0 {
+		// Build observed config only if the inactivity timeout value is set in OAuth cluster config.
+		observedConfig = buildUnstructuredTokenConfig(observedAccessTokenInactivityTimeout, tokenInactivityTimeoutPath)
+	}
+
+	if existingAccessTokenInactivityTimeoutFloat != observedAccessTokenInactivityTimeout {
+		recorder.Eventf("ObserveTokenConfig", "%s changed from %v to %v", fieldAccessTokenInactivityTimeoutSeconds,
+			existingAccessTokenInactivityTimeoutFloat,
+			observedAccessTokenInactivityTimeout)
+	}
+
+	return observedConfig, errs
+}
+
+func buildUnstructuredTokenConfig(val interface{}, fields []string) map[string]interface{} {
+	unstructuredConfig := map[string]interface{}{}
+
+	if err := unstructured.SetNestedField(unstructuredConfig, val, fields...); err != nil {
+		// SetNestedField can return only return an error if the fields are not present in map.
```

Comment:
double "return"

I don't think the sentence is correct. Instead `SetNestedField` returns an error if they *are* present, but are of wrong kind (i.e. of float type if an object is expected).

Reply:
Done.